pandas/core/index.py

```diff
@@ -1163,7 +1163,13 @@ def shift(self, periods=1, freq=None):
             return self
 
         offset = periods * freq
```

Comment:
Unrelated to the original issue, but it is nice to raise understandable error when ``freq`` is ``None``.